sled-agent/src/services.rs

```diff
@@ -815,127 +796,8 @@ impl ServiceManager {
             return Ok(Some(ledger));
         }
 
-        // Now look for the ledger used by previous versions.  If we find it,
```

Comment:
I kind of feel like we should try to remove any old-format ledger that we find here.  The discussion at L894-L936 is a good summary of the tradeoffs, except that now we know we're in the case where this ledger can't possibly be useful.  It also almost certainly can't be dangerous, either -- just potentially confusing to humans.